editor/src/clj/editor/fxui.clj

```diff
@@ -451,6 +468,40 @@
                                        :error "text-area-error"
                                        :borderless "text-area-borderless"))))
 
+(def ^:private ext-with-popup-on-props
+  (fx/make-ext-with-props
+    {:on (fx.prop/make
+           (fx.mutator/adder-remover
+             (fn [^Popup popup [^Node on anchor-x anchor-y]]
+               (.show popup on (double anchor-x) (double anchor-y)))
+             (fn [^Popup popup _]
+               (.hide popup)))
+           (tuple-lifecycle fx.lifecycle/dynamic fx.lifecycle/scalar fx.lifecycle/scalar))}))
+
+(defn with-popup
+  "Helper popup lifecycle that adds a managed popup to :desc node
+
+  Supported props:
+    :desc        node desc that will show a popup
+    :showing     whether the popup is showing
+    :anchor-x    screen anchor x
+    :anchor-y    screen anchor y
+    ...the rest of :popup props"
+  [{:keys [desc showing anchor-x anchor-y]
+    :or {anchor-x 0.0
+         anchor-y 0.0}
+    :as props}]
+  {:fx/type fx/ext-let-refs
+   :refs {::on desc}
+   :desc {:fx/type fx/ext-let-refs
+          :refs {::popup {:fx/type ext-with-popup-on-props
+                          :props (when showing
+                                   {:on [{:fx/type fx/ext-get-ref :ref ::on} anchor-x anchor-y]})
+                          :desc (-> props
+                                    (dissoc :desc :showing :anchor-x :anchor-y)
+                                    (assoc :fx/type :popup))}}
```

Comment:
Should we use `cljfx.fx.popup/lifecycle` here instead of the `:popup` keyword?

Reply:
Ah good catch, forgot about that!